Ukol_3/cpp/linked_list.cpp

```diff
@@ -0,0 +1,193 @@
+#include <iostream>
+#include <string>
+
+using namespace std;
+
+struct Node
+{
+    int data;
+    Node *next;
+};
+
+// Funkce pro vytvoření nového uzlu
+Node *createNode(int data)
+{
+    Node* node = new Node();
+    node->data = data;
+    node->next = nullptr;
+}
+
+// Funkce pro vložení uzlu na začátek seznamu
+void insertAtBeginning(Node **head, int data)
+{
+    Node* newNode = createNode(data);
+    //do nově vytvořeného uzlu dám ukazatel na 'head' , vložíme ho na začátek seznamu
+    newNode->next = *head;
+    *head = newNode;
+}
+
+// Funkce pro vložení uzlu na konec seznamu
+void insertAtEnd(Node **head, const int data)
+{
+    Node* newNode = createNode(data);
+    // pokud je seznam prázdný, pouze vložím nově vytvořeny uzel
+    if (*head == nullptr){
+        *head = newNode;
+    }else{
+        Node* tmp = *head; // vytvořim si dočasný uzel, kopii 'head'
+    // dokuď nejsem na konci seznamu, posouvám si pozici 'tmp'
+        while (tmp->next!= nullptr){
+            tmp = tmp->next;
+        }
+        tmp->next = newNode;// 'tmp' ma pozici na konci seznamu, vlozim tam noví uzel
+    }
+}
+
+// Funkce pro vložení na index
+void insertAtIndex(Node **head, int data, int index)
```

Comment:
Tato funkce nefunguje správně pokud chceme vložit hodnotu na 0 index tedy na začátek linked listu.